File: backend/app/api/schemas/evidence.py
"""Pydantic schemas for Evidence API (TP-1003)."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class EvidenceResponse(BaseModel):
    id: str
    session_id: str
    evidence_type: str
    status: str
    original_filename: str | None
    mime_type: str | None
    file_size_bytes: int | None
    checksum_sha256: str | None
    market_timestamp: datetime | None
    uploaded_at: datetime
    caption: str | None
    supersedes_evidence_id: str | None

    model_config = {"from_attributes": True}


class EvidenceListResponse(BaseModel):
    evidence: list[EvidenceResponse]
    total: int


class EvidenceErrorResponse(BaseModel):
    code: str
    message: str

File: frontend/src/types/evidence.ts
/** Evidence metadata */
export interface EvidenceItem {
  id: string;
  session_id: string;
  evidence_type: string;
  status: string;
  original_filename: string | null;
  mime_type: string | null;
  file_size_bytes: number | null;
  checksum_sha256: string | null;
  market_timestamp: string | null;
  uploaded_at: string;
  caption: string | null;
  supersedes_evidence_id: string | null;
}

/** GET /api/trade-sessions/{id}/evidence */
export interface ListEvidenceResponse {
  evidence: EvidenceItem[];
  total: number;
}
